util/resource/resource.go

```diff
@@ -17,61 +17,54 @@ limitations under the License.
 // Package resource implements resource utilites.
 package resource
 
-import "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+import (
+	"sort"
 
-var defaultCreatePriorities = []string{
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var priorityMap = map[string]int{
 	// Namespaces go first because all namespaced resources depend on them.
-	"Namespace",
+	"Namespace": -100,
```

Comment:
It seems a bit strange that the kind with the highest priority has the lowest priority value

Reply:
Apart from that, nice improvement!

Reply:
@kokes Can we have a quick follow-up PR to align priority values to priority?

Reply:
This is so that values are sorted naturally without having to fiddle with anything. I thought about giving these positive values, which would then require the `Less` function to have the inequality check reverted... I thought it'd be a bit confusing. But if it sounds confusing to have it this way, fine, we can flip it, the tests should cover this nicely.

Reply:
I just thought if the map is the "priority map" then I would expect the kind with the highest priority to come first not last

Reply:
If it would be called the orderMap it's perfectly fine I guess :)

Reply:
Like this? https://github.com/kubernetes-sigs/cluster-api/compare/main...kokes:sort_order_naming?expand=1

I kept `priorityLess`, even though that might sound confusing - but it's for sorting by priority and it's the `Less` function for native Go sorting, so I thought I'd get away with that.

Reply:
Yeah let's go with that